src/H5Dchunk.c

```diff
@@ -764,9 +764,6 @@ H5D__chunk_set_sizes(H5D_t *dset)
     /* Sanity checks */
     assert(dset);
 
-    /* Increment # of chunk dimensions, to account for datatype size as last element */
-    dset->shared->layout.u.chunk.ndims++;
```

Comment:
It may be useful to add an assertion here that `ndims > 0`. Previously it was inherent to the code but now that this increment has been moved out of this function, a future re-org could mean that the below code ends up performing negative indexing.